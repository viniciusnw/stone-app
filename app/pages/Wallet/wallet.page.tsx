import React from "react";
import { Text, View } from "react-native";
import {
  CreditCard,
  Profile as ProfileComp,
  Wallet as WalletComp,
} from "@Components";
import { useNavigation } from "@react-navigation/native";

export type CardProps = {
  name: string;
  number: string;
  brand: "visa" | "mastercard" | "santander";
};

export default function Wallet(props: any) {
  const navigation = useNavigation();

  const user = {
    src: "https://i.pinimg.com/280x280_RS/39/fd/09/39fd09a3f0962e5b7b4a3ccbf5a61057.jpg",
    name: "Bruce Wayne",
    rule: "Wayne Interprise",
  };

  const card: CardProps = {
    name: "Bruce Wayne",
    number: "**** **** **** 1234",
    brand: "visa",
  };

  const card1: CardProps = {
    name: "Dick Grayson",
    number: "**** **** **** 1234",
    brand: "mastercard",
  };

  React.useEffect(
    () =>
      props.setState({
        bottomBar: {
          menu: [
            {
              icon: "wallet",
              label: "Carteira",
              onPress: () => true,
              primary: false,
            },
            {
              icon: "home",
              label: "Home",
              onPress: () => props.goTo("Home"),
              primary: true,
            },
            {
              icon: "logout",
              label: "Logout",
              onPress: () => true,
              primary: false,
            },
          ],
        },
        topBar: {
          title: "Wallet",
          back: navigation.goBack,
        },
      }),
    []
  );

  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <WalletComp value="R$ 10.000,00" qty={200} />

      <View style={{ padding: 10 }}>
        <Text style={{ marginBottom: 15 }}>Meus cartoes:</Text>
        <CreditCard {...card} />
        <CreditCard {...card1} />
      </View>
    </View>
  );
}
