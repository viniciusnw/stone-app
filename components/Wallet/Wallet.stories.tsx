import React from "react";
import { Wallet, StorybookWrapper } from "@Components";


const sample = {
  value: "R$1350,50",
  qty: 8,
};

export const Default = {};

export default {
  title: "Wallet",
  component: (props: any) => (
    <StorybookWrapper>
      <Wallet {...sample} />
    </StorybookWrapper>
  ),
};
