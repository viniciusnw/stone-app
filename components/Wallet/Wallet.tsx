import React from 'react';
import styled from 'styled-components/native';
import { View, Text } from 'react-native';
import Icon from '@Components/Icon';

const StyledContainer = styled(View)`
  width: 100%;
  align-items: center;
  padding: 60px ${({ theme }) => theme.spacing.regular};
`;

const StyledValue = styled(Text)`
  font-size: 32px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.black};
`;

const StyledQty = styled(Text)`
  margin-top: 5px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.gray};
`;

const StyledIcon = styled(View)`
  width: 52px;
  height: 52px;
  align-items: center;
  justify-content: center;
  border: 1px solid ${({ theme }) => theme.color.border};
  border-radius: 100px;
  margin-bottom: ${({ theme }) => theme.spacing.regular};
`;

const Wallet = ({ value, qty, ...props }: any) => {
  return (
    <StyledContainer {...props}>
      <StyledIcon>
        <Icon icon="wallet" size={26} />
      </StyledIcon>
      <StyledValue>{value}</StyledValue>
      <StyledQty>{`${qty} pedido${qty > 1 ? 's' : ''}`}</StyledQty>
    </StyledContainer>
  );
};

export default Wallet;
