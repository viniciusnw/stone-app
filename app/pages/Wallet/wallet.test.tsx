import React from "react";
import renderer from "react-test-renderer";
import { lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import Wallet from "./wallet.page";
import "jest-styled-components";
import Wrapper from "@/app/navigation/wrapper.navigator";

jest.useFakeTimers();

jest.mock("@react-navigation/native", () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
    useRoute: () => ({
      params: {
        productId: 1,
      },
    }),
  };
});

test("WalletWrapper", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <Wrapper Page={Wallet} header bottomBarType={null} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Wallet", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={lightTheme}>
        <Wallet setState={jest.fn} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
