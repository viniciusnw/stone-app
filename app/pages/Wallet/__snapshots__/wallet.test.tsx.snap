// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "paddingBottom": 60,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 60,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 100,
          "borderBottomRightRadius": 100,
          "borderColor": "#ddd",
          "borderStyle": "solid",
          "borderTopLeftRadius": 100,
          "borderTopRightRadius": 100,
          "borderWidth": 1,
          "height": 52,
          "justifyContent": "center",
          "marginBottom": 15,
          "width": 52,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 26,
            },
            undefined,
            {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        îˆ§
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#111111",
          "fontSize": 32,
          "fontWeight": "700",
        }
      }
    >
      R$ 10.000,00
    </Text>
    <Text
      style={
        {
          "color": "#666666",
          "fontWeight": "700",
          "marginTop": 5,
        }
      }
    >
      200 pedidos
    </Text>
  </View>
  <View
    style={
      {
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "marginBottom": 15,
        }
      }
    >
      Meus cartoes:
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#ddd",
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 15,
          "paddingBottom": 0,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 52,
                "width": 52,
              }
            }
          />
        </View>
        <View
          style={
            {
              "borderColor": "transparent",
              "borderLeftColor": "#ddd",
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginLeft": 15,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              {
                "color": "#666666",
                "fontSize": 18,
                "fontWeight": "700",
                "marginBottom": 5,
              }
            }
          >
            Bruce Wayne
          </Text>
          <Text
            style={
              {
                "color": "#666666",
                "fontSize": 13,
                "fontWeight": "700",
              }
            }
          >
            **** **** **** 1234
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#ddd",
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 15,
          "paddingBottom": 0,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            {
              "justifyContent": "center",
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "height": 52,
                "width": 52,
              }
            }
          />
        </View>
        <View
          style={
            {
              "borderColor": "transparent",
              "borderLeftColor": "#ddd",
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginLeft": 15,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              {
                "color": "#666666",
                "fontSize": 18,
                "fontWeight": "700",
                "marginBottom": 5,
              }
            }
          >
            Dick Grayson
          </Text>
          <Text
            style={
              {
                "color": "#666666",
                "fontSize": 13,
                "fontWeight": "700",
              }
            }
          >
            **** **** **** 1234
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`WalletWrapper 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ddd",
        "borderColor": "transparent",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "width": 30,
        }
      }
    />
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#111111",
          "fontSize": 20,
          "fontWeight": "700",
          "maxWidth": 200,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "FredokaOne-Regular",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0,168,104)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Stone
          </Text>
          <Text
            style={
              {
                "color": "rgb(10,245,61)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Ton :)
          </Text>
        </Text>
      </View>
    </Text>
    <View
      style={
        {
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "paddingBottom": 60,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 60,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomLeftRadius": 100,
            "borderBottomRightRadius": 100,
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderTopLeftRadius": 100,
            "borderTopRightRadius": 100,
            "borderWidth": 1,
            "height": 52,
            "justifyContent": "center",
            "marginBottom": 15,
            "width": 52,
          }
        }
      >
        <Text />
      </View>
      <Text
        style={
          {
            "color": "#111111",
            "fontSize": 32,
            "fontWeight": "700",
          }
        }
      >
        R$ 10.000,00
      </Text>
      <Text
        style={
          {
            "color": "#666666",
            "fontWeight": "700",
            "marginTop": 5,
          }
        }
      >
        200 pedidos
      </Text>
    </View>
    <View
      style={
        {
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "marginBottom": 15,
          }
        }
      >
        Meus cartoes:
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 15,
            "paddingBottom": 0,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              {
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 52,
                  "width": 52,
                }
              }
            />
          </View>
          <View
            style={
              {
                "borderColor": "transparent",
                "borderLeftColor": "#ddd",
                "borderStyle": "solid",
                "borderWidth": 1,
                "marginLeft": 15,
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#666666",
                  "fontSize": 18,
                  "fontWeight": "700",
                  "marginBottom": 5,
                }
              }
            >
              Bruce Wayne
            </Text>
            <Text
              style={
                {
                  "color": "#666666",
                  "fontSize": 13,
                  "fontWeight": "700",
                }
              }
            >
              **** **** **** 1234
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#ddd",
            "borderStyle": "solid",
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 15,
            "paddingBottom": 0,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              {
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                {
                  "height": 52,
                  "width": 52,
                }
              }
            />
          </View>
          <View
            style={
              {
                "borderColor": "transparent",
                "borderLeftColor": "#ddd",
                "borderStyle": "solid",
                "borderWidth": 1,
                "marginLeft": 15,
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#666666",
                  "fontSize": 18,
                  "fontWeight": "700",
                  "marginBottom": 5,
                }
              }
            >
              Dick Grayson
            </Text>
            <Text
              style={
                {
                  "color": "#666666",
                  "fontSize": 13,
                  "fontWeight": "700",
                }
              }
            >
              **** **** **** 1234
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
